querybook/server/tasks/run_datadoc.py

```diff
@@ -117,13 +117,14 @@ def run_datadoc_with_config(
 @celery.task(bind=True)
 def _start_query_execution_task(
     self,
-    previous_query_status,
+    previous_query_result,
     cell_id,
     query_execution_params,
     data_doc_id,
 ):
+    previous_query_status, previous_query_execution_id = previous_query_result
     if previous_query_status != QueryExecutionStatus.DONE.value:
-        raise Exception(GENERIC_QUERY_FAILURE_MSG)
+        raise Exception(get_datadoc_error_message(previous_query_execution_id))
```

Comment:
@czgu do you know why we raise the exception in the next query cell run task, instead of just raising it at the end of the `run_query` task? In which case we dont need to pass over the previous query result over and can throw the error message directly